datafusion/optimizer/src/decorrelate_scalar_subquery.rs

```diff
@@ -702,4 +727,31 @@ mod tests {
         assert_optimized_plan_eq(&DecorrelateScalarSubquery::new(), &plan, expected);
         Ok(())
     }
+
+    /// Test for non-correlated scalar subquery with no filters
```

Comment:
Should we rename this if it also does uncorrelated subqueries now? Maybe `scalar_subquery_to_join`?

Reply:
You mean rename the whole rule? Probably so .. good idea. You want to do that as part of this issue?

Reply:
I don't want to, but we probably should. Maybe we can just doc it for now with a TODO and address it in a future PR?

Reply:
It touches a bunch of code, but yes, your suggested name is much more clear. I'll do a separate PR for it this week if this gets merged in.